aiohttp/web_fileresponse.py

```diff
@@ -179,31 +184,72 @@ def _get_file_path_stat_encoding(
                 # Do not follow symlinks and ignore any non-regular files.
                 st = compressed_path.lstat()
```

Comment:
Why not `open` before the `lstat` and then use `os.lstat` on the file object?

Reply:
Because it could be a socket and each operating system will handle it slightly differently and raise different errno, some of which are not mapped in Cpython